From b00fc4676127cd28d3fb2a14874d363a19ba1475 Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Sat, 21 Nov 2020 19:19:48 -0800
Subject: [PATCH 093/171] QMutex: Work around ICC bug in dealing with constexpr
 constructor

The bug report says it also happens in some MSVC builds, but no one has
so far been able to reproduce it or give us a log with more details, so
not working around the issue for MSVC. ICC, howevere, is known to have a
lot of parsing issues, so I'm not surprised and don't feel a problem
working around them.

This fix is not applicable for Qt 6.

Fixes: QTBUG-85644
Change-Id: If51855da004b4f3fbf43fffd1649b64e05cb91cc
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
---
 src/corelib/thread/qmutex.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/corelib/thread/qmutex.h b/src/corelib/thread/qmutex.h
index 93c4bf23e8..73c9e00663 100644
--- a/src/corelib/thread/qmutex.h
+++ b/src/corelib/thread/qmutex.h
@@ -129,7 +129,7 @@ private:
 class Q_CORE_EXPORT QMutex : public QBasicMutex
 {
 public:
-#if defined(Q_COMPILER_CONSTEXPR)
+#if defined(Q_COMPILER_CONSTEXPR) && !defined(Q_CC_INTEL)
     constexpr QMutex() = default;
 #else
     QMutex() { d_ptr.storeRelaxed(nullptr); }
-- 
2.31.1

